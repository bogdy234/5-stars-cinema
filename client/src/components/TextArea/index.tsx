import React, {FC, ReactElement} from 'react';

interface TextAreaProps {
    placeholder: string;
    value: string;
    label?: string;
    onChange: (newValue: string) => void;
    className?: string;
}

const TextArea: FC<TextAreaProps> = ({placeholder, value, label, onChange, className}: TextAreaProps): ReactElement =>
    (
        <div className={className}>
            <label>
                {label}
                <textarea
                    placeholder={placeholder}
                    value={value}
                    onChange={(e) => onChange(e.target.value)}
                    className="w-full h-32 pl-4 pt-2 resize-none"
                />
            </label>
        </div>
    );

export default TextArea;
