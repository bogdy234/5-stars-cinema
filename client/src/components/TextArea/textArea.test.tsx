import renderer from "react-test-renderer";
import TextArea from "./index";

it("renders correctly", () => {
  const tree = renderer
    .create(<TextArea placeholder={""} value={""} onChange={jest.fn()} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
