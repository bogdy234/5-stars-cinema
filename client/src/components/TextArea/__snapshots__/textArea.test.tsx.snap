// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <label>
    <textarea
      className="w-full h-32 pl-4 pt-2 resize-none"
      onChange={[Function]}
      placeholder=""
      value=""
    />
  </label>
</div>
`;
